runtime/web_worker.rs

```diff
@@ -165,11 +166,32 @@ impl WebWorker {
     worker_id: u32,
     options: &WebWorkerOptions,
   ) -> Self {
+    let extensions: Vec<Extension> = vec![
+      // Web APIs
+      deno_webidl::init(),
+      deno_console::init(),
+      deno_url::init(),
+      deno_web::init(),
+      deno_fetch::init::<Permissions>(
+        options.user_agent.clone(),
+        options.ca_data.clone(),
+      ),
+      deno_websocket::init::<Permissions>(
+        options.user_agent.clone(),
+        options.ca_data.clone(),
+      ),
+      deno_crypto::init(options.seed),
+      deno_webgpu::init(options.unstable),
+      // Metrics
+      metrics::init(),
```

Comment:
Not entirely sure about metrics being a middleware - I discussed it with @ry, concretely `metrics` do not properly track usage of zero copy buffers - eg. in case of "read" op metrics measure the size of buffer, but in reality it can only be filed with a few bytes (but the op will count the size of buffer instead). Maybe skip it for this PR?

Reply:
@bartlomieju I agree that's an issue, but isn't that separate from it being middleware ?
(that issue exists on `main`)

Without that middleware, it will break metrics for op_crate ops (ops declared via this new extension system). I added a test that checks that metrics work for `op_url_parse` (op_crate metrics weren't tested at all, so it would have caused an unexpected fail/crash for users)

Reply:
That's correct, but using metrics as a middleware I don't see a way how collection of metrics could be customized per op basis

Reply:
@bartlomieju Metrics are exposed via `Deno.metrics` in stable, so we have to keep the current middleware collection until 2.0 at least.

Reply:
@bartlomieju Was it ever customized before ? I might have missed that, I don't remember seeing anything to that effect.

It sounds like this is an orthogonal fix/change, that should happen in a follow-up PR and would have had to change with the new op-ABI anyway.

If you remove op-middleware and the metrics middleware right now, you'll cause breaking changes for users, unless you verbosely re-wrap all the ops but that kinda defeats the whole point of extension modularity.